### DISCUSSION TITLE
Is there a way of showing Arabic words on the map, on widgets, and/or shapes?

### DISCUSSION THREAD

**Discussion by uSafi** [Mapsui/Mapsui#2654] [Q&A]
So far, the Arabic characters appear as rectangular box glyphs. 
Could there be code base encoding issues, or a way to set it right?.

**Comment by charlenni** [comment]
Do you use a font that is capable to displaying this characters?

**Reply by uSafi** [reply]
> Do you use a font that is capable to displaying this characters?

Just the default fonts set in MauiProgram.cs.
That's normally enough, unless one want to show something looks better.
Btw: I yet to know how to select fonts for Mapsui, but someday I'll get to the point where I should set custom fonts.

**Reply by charlenni** [reply]
The default fonts from Maui and Mapsui could be different. So you should look into, how to set the right font.

**Reply by uSafi** [reply]
> The default fonts from Maui and Mapsui could be different. So you should look into, how to set the right font.

And if i may to ask, how to set different font to Mapsui?
I've searched and asked AI, the road to do that requires steps like:

- Create a custom style: Derive a class from IStyle. Assign that style to an ILayer.Style or IFeature.Styles.
- Create a custom renderer: Derive a class from ISkiaStyleRenderer and implement the Draw method. In this method, you can use 
  SkiaSharp’s text rendering functions to draw text with your custom font.
- Register the association: Register the association of the custom style to the custom style renderer. This can be done as follows:
        mapControl.Renderer.StyleRenderers.Add(typeof(CustomStyle), new SkiaCustomStyleRenderer());
        
Note: I could finish my project by putting Pins with different colors on the map, the colors correspond to named legend with text outside the map (something like buttons outside the map with corresponding colors)
 but that's a work around that might be unnecessary if it turns out making Arabic text on the map is doable with ease.
 
 That's why I came here to ask.

**Reply by uSafi** [reply]
> The default fonts from Maui and Mapsui could be different. So you should look into, how to set the right font.

I'll look into the answer here by [pauldendulk](https://github.com/pauldendulk): https://github.com/Mapsui/Mapsui/discussions/2543#discussioncomment-8757093
And see if setting different font to Labels can be the answer so I can use Labels instead of Pins>
But I have my doubts, because of past experience when dealing with RTL languages.
